invoice/src/builder.rs

```diff
@@ -107,6 +105,15 @@ impl RgbInvoiceBuilder {
         Ok(self.set_amount_raw(amount))
     }
 
+    pub fn set_allocation_raw(mut self, allocation: impl Into<Allocation>) -> Self {
+        self.0.owned_state = InvoiceState::Data(NonFungible::RGB21(allocation.into()));
```

Comment:
I am afraid `into()` introduces random blinding. In `_raw` methods we need to explicitly provide blinding information, otherwise we won't be able to deterministically create state transitions (required in Lightning etc).

Reply:
Hi, 

I'm not sure if I understand what you mean, because in invoice creation, we generate blinding information in beneficiary instead of amount or allocation.